packages/liveblocks-react-ui/src/components/internal/AiComposer.tsx

```diff
@@ -0,0 +1,212 @@
+  AiComposerEditorProps,
+  AiComposerFormProps,
+  AiComposerSubmitMessage,
+} from "../../primitives/AiComposer/types";
+import { cn } from "../../utils/cn";
+import { Button } from "./Button";
+import { ShortcutTooltip, TooltipProvider } from "./Tooltip";
+
+/* -------------------------------------------------------------------------------------------------
+ * AiComposer
+ * -----------------------------------------------------------------------------------------------*/
+export interface AiComposerProps
+  extends Omit<ComponentProps<"form">, "defaultValue"> {
+  /**
+   * The composer's initial value.
+   */
+  defaultValue?: string;
+
+  /**
+   * The event handler called when the composer is submitted.
+   */
+  onComposerSubmit?: (
+    message: AiComposerSubmitMessage,
+    event: FormEvent<HTMLFormElement>
+  ) => void;
+
+  /**
+   * @internal
+   * The event handler called after the composer is submitted.
+   */
+  onComposerSubmitted?: (message: AiChatMessage) => void;
+
+  /**
+   * Whether the composer is disabled.
+   */
+  disabled?: AiComposerFormProps["disabled"];
+
+  /**
+   * Whether to focus the composer on mount.
+   */
+  autoFocus?: AiComposerEditorProps["autoFocus"];
+
+  /**
+   * Override the component's strings.
+   */
+  overrides?: Partial<GlobalOverrides & AiComposerOverrides>;
+
+  /**
+   * The ID of the chat the composer belongs to.
+   */
+  chatId: string;
+
+  /**
+   * The ID of the copilot to use to send the message.
+   */
+  copilotId?: CopilotId;
+
+  /**
+   * @internal
+   */
+  branchId?: MessageId;
+
+  /**
+   * @internal
+   */
+  stream?: boolean;
+}
+
+function AiComposerAction({
+  overrides,
+}: {
+  overrides?: AiComposerProps["overrides"];
+}) {
+  const { canAbort } = useAiComposer();
+  const $ = useOverrides(overrides);
+
+  const preventDefault = useCallback((event: SyntheticEvent) => {
+    event.preventDefault();
+  }, []);
+
+  const stopPropagation = useCallback((event: SyntheticEvent) => {
+    event.stopPropagation();
+  }, []);
+
+  return canAbort ? (
+    <ShortcutTooltip content={$.AI_COMPOSER_ABORT}>
+      <AiComposerPrimitive.Abort asChild>
+        <Button
+          onPointerDown={preventDefault}
+          onClick={stopPropagation}
+          className="lb-ai-composer-action"
+          variant="secondary"
+          aria-label={$.AI_COMPOSER_ABORT}
+          icon={<StopIcon />}
+        />
+      </AiComposerPrimitive.Abort>
+    </ShortcutTooltip>
+  ) : (
+    <ShortcutTooltip content={$.AI_COMPOSER_SEND} shortcut="Enter">
+      <AiComposerPrimitive.Submit asChild>
+        <Button
+          onPointerDown={preventDefault}
+          onClick={stopPropagation}
+          className="lb-ai-composer-action"
+          variant="primary"
+          aria-label={$.AI_COMPOSER_SEND}
+          icon={<SendIcon />}
+        />
+      </AiComposerPrimitive.Submit>
+    </ShortcutTooltip>
+  );
+}
+
+export const AiComposer = forwardRef<HTMLFormElement, AiComposerProps>(
```

Comment:
Should we consider renaming this to `InternalAiComposer` to make it more explicit? The fact that you can now use an `AiComposer` from either `src/components/internal/` and `src/components/primitives/` is a bit confusing to me, and it's not obvious to me if you're reading code that is pretty far away from the import statement.

Reply:
For now we could, but one possibility is that we'll make `"Internal"AiComposer` publicly usable as a standalone component from `@liveblocks/react-ui` like Comment's `Composer`, and at the same time we'll also offer `AiComposer` from `@liveblocks/react-ui/primitives` but they're two different things. We technically have this issue already with Comments: `Composer` from `@liveblocks/react-ui` isn't the same as `Composer` from `@liveblocks/react-ui/primitives`, one is a pre-built component and the other is a few unstyled components grouped as a "primitive", same for `Comment`, etc.